etl/glean_etl.py

```diff
@@ -59,9 +59,25 @@ def _get_annotation(annotations_index, origin, item_type, identifier=None):
 def _incorporate_annotation(item, item_annotation, app=False, full=False):
     incorporated = dict(item, has_annotation=len(item_annotation) > 0)
 
-    # handle tags, which are in a slightly different format than other types
-    # of annotations (and we want to apply to both full and non-full)
-    if not app and item_annotation.get("tags"):
+    if app:
+        # app annotations have some special properties
+        if item_annotation.get("logo"):
+            # the logo is dowloaded locally SVG elsewhere
+            incorporated.update(
+                {"logo": f"/data/{item['app_name']}/" + _get_logo_filename(item_annotation["logo"])}
+            )
+
+        if item_annotation.get("featured"):
+            incorporated["featured"] = True
+
+        # we use the `app_tags` property to disambiguate between the tags
+        # that are a property of an application vs. the tags that it has
+        # on other things
```

Comment:
I realize all of this logic is kind of confusing, but I'm not sure how to make it better right now.